bug: load_packed_bytes is under constrained

## Bug Report

PoC

```
diff --git a/src/utils/utils.cairo b/src/utils/utils.cairo
index da451281..ce617f07 100644
--- a/src/utils/utils.cairo
+++ b/src/utils/utils.cairo
@@ -1178,7 +1178,8 @@ namespace Helpers {
         // Put byte in output and assert that 0 <= byte < bound
         // See math.split_int
         %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
+            memory[ids.output] = res = 0x12
+            # memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
             assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
         %}
         tempvar a = [output];
```

makes test `poetry run pytest tests/src/ -m "not NoCI" --log-cli-level=INFO -n logical -vv -k test_should_load_packed_bytes` fail with

```
    def test_should_load_packed_bytes(self, cairo_run):
        bytes = random.randbytes(100)
        packed_bytes = pack_calldata(bytes)
        output = cairo_run("test__load_packed_bytes", data=packed_bytes)
>       assert output == list(bytes)
E       assert [18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18] == [106, 183, 253, 213, 34, 28, 138, 66, 73, 205, 235, 17, 121, 68, 136, 56, 143, 188, 108, 18, 152, 236, 156, 165, 127, 94, 18, 77, 152, 221, 172, 89, 233, 49, 162, 111, 101, 80, 41, 46, 63, 122, 160, 15, 110, 82, 238, 128, 134, 233, 149, 119, 10, 185, 20, 10, 110, 60, 179, 152, 112, 249, 213, 25, 0, 215, 6, 179, 129, 250, 252, 252, 72, 173, 42, 100, 46, 251, 8, 51, 197, 23, 152, 66, 190, 71, 202, 91, 55, 171, 134, 231, 203, 6, 74, 187, 2, 22, 96, 120]
```

--- a/src/utils/utils.cairo
+++ b/src/utils/utils.cairo
@@ -1206,11 +1206,18 @@ namespace Helpers {
 
         jmp cond if remaining_bytes != 0;
 
+        with_attr error_message("Value is not empty") {
+            assert value = 0;
+        }
+
         let bytes = cast([fp], felt*);
         return (bytes=bytes);
 
         cond:
         jmp body if count != 0;
+        with_attr error_message("Value is not empty") {
+            assert value = 0;
+        }
         jmp read;
     }
 

--- a/src/utils/utils.cairo
+++ b/src/utils/utils.cairo
@@ -1120,13 +1120,12 @@ namespace Helpers {
         alloc_locals;
 
         let (local bytes: felt*) = alloc();
-        local bound = 256;
-        local base = 256;
-
         if (bytes_len == 0) {
             return (bytes=bytes);
         }
 
+        local bound = 256;
+        local base = 256;
         let (local chunk_counts, local remainder) = unsigned_div_rem(bytes_len, BYTES_PER_FELT);
 
         tempvar remaining_bytes = bytes_len;
@@ -1145,8 +1144,8 @@ namespace Helpers {
 
         tempvar value = input[index];
 
-        let remainder = [fp + 4];
         let chunk_counts = [fp + 3];
+        let remainder = [fp + 4];
         tempvar remaining_chunk = chunk_counts - index;
         jmp full_chunk if remaining_chunk != 0;
         tempvar count = remainder;
@@ -1169,9 +1168,9 @@ namespace Helpers {
         let value = [ap - 2];
         let count = [ap - 1];
 
-        let base = [fp + 1];
-        let bound = [fp + 2];
         let bytes = cast([fp], felt*);
+        let bound = [fp + 1];
+        let base = [fp + 2];
 
         tempvar offset = (index - 1) * BYTES_PER_FELT + count - 1;
         let output = bytes + offset;
@@ -1209,7 +1208,6 @@ namespace Helpers {
         with_attr error_message("Value is not empty") {
             assert value = 0;
         }
-
         let bytes = cast([fp], felt*);
         return (bytes=bytes);
 

--- a/src/utils/utils.cairo
+++ b/src/utils/utils.cairo
@@ -1146,6 +1146,7 @@ namespace Helpers {
 
         let chunk_counts = [fp + 3];
         let remainder = [fp + 4];
+
         tempvar remaining_chunk = chunk_counts - index;
         jmp full_chunk if remaining_chunk != 0;
         tempvar count = remainder;

--- a/tests/src/utils/test_utils.py
+++ b/tests/src/utils/test_utils.py
@@ -168,7 +168,7 @@ class TestLoadPackedBytes:
         output = cairo_run("test__load_packed_bytes", data=packed_bytes)
         assert output == list(bytes)
 
-    def test_should_rasise_zellic_issue_1283_load_packed_bytes(
+    def test_should_raise_zellic_issue_1283_load_packed_bytes(
         self, cairo_program, cairo_run
     ):
         bytes = random.randbytes(100)

--- a/tests/src/utils/test_utils.py
+++ b/tests/src/utils/test_utils.py
@@ -177,8 +177,8 @@ class TestLoadPackedBytes:
             patch_hint(
                 cairo_program,
                 "memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base\nassert res < ids.bound, f'split_int(): Limb {res} is out of range.'",
-                "memory[ids.output] = res = 0x12\nassert res < ids.bound, f'split_int(): Limb {res} is out of range.'",
+                "memory[ids.output] = res = 0x12",
             ),
+            cairo_error(message="Value is not empty"),
         ):
-            with cairo_error(message="Value is not empty"):
-                cairo_run("test__load_packed_bytes", data=packed_bytes)
+            cairo_run("test__load_packed_bytes", data=packed_bytes)

--- a/tests/src/utils/test_utils.py
+++ b/tests/src/utils/test_utils.py
@@ -7,7 +7,9 @@ from hypothesis import given, settings
 from hypothesis import strategies as st
 
 from kakarot_scripts.utils.kakarot import get_contract
+from tests.utils.errors import cairo_error
 from tests.utils.helpers import pack_calldata
+from tests.utils.hints import patch_hint
 
 
 @pytest.mark.parametrize(
@@ -165,3 +167,18 @@ class TestLoadPackedBytes:
         packed_bytes = pack_calldata(bytes)
         output = cairo_run("test__load_packed_bytes", data=packed_bytes)
         assert output == list(bytes)
+
+    def test_should_rasise_zellic_issue_1283_load_packed_bytes(
+        self, cairo_program, cairo_run
+    ):
+        bytes = random.randbytes(100)
+        packed_bytes = pack_calldata(bytes)
+        with (
+            patch_hint(
+                cairo_program,
+                "memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base\nassert res < ids.bound, f'split_int(): Limb {res} is out of range.'",
+                "memory[ids.output] = res = 0x12\nassert res < ids.bound, f'split_int(): Limb {res} is out of range.'",
+            ),
+        ):
+            with cairo_error(message="Value is not empty"):
+                cairo_run("test__load_packed_bytes", data=packed_bytes)

